### PR TITLE
code fragment & english syntax and meaning

### PR DESCRIPTION
**Pull request by CakeCrusher**
some english sentences did not make much sense so I tweaked them a bit.

### CODE REVIEW

File: chapters/en/chapter8/4.mdx [hunk 1]
```diff
@@ -185,7 +185,7 @@ trainer.train_dataset[0].keys()
 dict_keys(['attention_mask', 'hypothesis', 'idx', 'input_ids', 'label', 'premise'])
 ```
 
-Note that the keys that don't correspond to inputs accepted by the model will be automatically discarded, so here we will only keep `input_ids`, `attention_mask`, and `label` (which will be renamed `labels`). To double-check the model signature, you can print the class of your model, then go check its documentation:
+Note that the keys that don't correspond to inputs accepted by the model as some will be automatically discarded, so here we will only keep `input_ids`, `attention_mask`, and `label` (which will be renamed `labels`). To double-check the model signature, you can print the class of your model, then go check its documentation:
```
Comment (lewtun) [inline chapters/en/chapter8/4.mdx:188]: The original text is actually correct, and adding "as some" doesn't quite capture the original meaning. Here's an alternative formulation that might be a bit clearer:

> In most cases, your model will only accept a subset of the keys (i.e. columns) that are present in the dataset. To prevent errors during training and inference, the `Trainer` will thus discard any key that isn't present in the model's forward pass. In our case, this means only `input_ids`, `attention_mask`, and `label` (which will be renamed `labels`) will remain.

File: chapters/en/chapter8/4.mdx [hunk 2]
```diff
@@ -200,7 +200,7 @@ So in our case, we can check the parameters accepted on [this page](https://hugg
 We have checked that the input IDs are correct by decoding them. Next is the `attention_mask`:
 
 ```py
-tokenizer.decode(trainer.train_dataset[0]["attention_mask"])
+trainer.train_dataset[0]["attention_mask"]
```
Comment (lewtun) [inline chapters/en/chapter8/4.mdx:203]: Good catch!

File: chapters/en/chapter8/7.mdx [hunk 3]
```diff
@@ -192,7 +192,7 @@ Which of the following might be a good choice for the title of a forum topic to
 		},
 		{
 			text: "It allows the maintainers to know whether you're running code on a GPU or CPU.",
-			explain: "Correct! As we've seen in this chapter, errors on GPUs and CPUs can quite different in flavor, and knowing which hardware you're using can help focus the maintainers' attention. But this isn't the only benefit...",
+			explain: "Correct! As we've seen in this chapter, code ran on GPUs or CPUs may produce diffferent results, and knowing which hardware you're using can help focus the maintainers' attention. But this isn't the only benefit...",
```
Comment (lewtun) [inline chapters/en/chapter8/7.mdx:195]:
```suggestion
			explain: "Correct! As we've seen in this chapter, code ran on GPUs or CPUs may produce diffferent results or errors, and knowing which hardware you're using can help focus the maintainers' attention. But this isn't the only benefit...",
```

**Review by lewtun — COMMENTED** [review]
Thanks for the tweaks @CakeCrusher ! I think one of them needs to be replaced with a larger rewording of the paragraph - let me know what you think

File: chapters/en/chapter8/4.mdx (hunk 1 above)
Comment (CakeCrusher) [reply chapters/en/chapter8/4.mdx:188]: I see now, I misread it.

```suggestion
Note that the keys that don't correspond to inputs accepted by the model will be automatically discarded, so here we will only keep `input_ids`, `attention_mask`, and `label` (which will be renamed `labels`). To double-check the model signature, you can print the class of your model, then go check its documentation:
```

Comment (CakeCrusher) [reply chapters/en/chapter8/4.mdx:188]: I see now, I misread it.

```suggestion
Note that the keys that don't correspond to inputs accepted by the model will be automatically discarded, so here we will only keep `input_ids`, `attention_mask`, and `label` (which will be renamed `labels`). To double-check the model signature, you can print the class of your model, then go check its documentation:
```

**Comment by CakeCrusher** [comment]
@lewtun suggestions have been applied

**Review by lewtun — APPROVED** [review]
Thanks, this looks great!
